Shared/Samples/Show grid/ShowGridView.swift

```diff
@@ -0,0 +1,318 @@
+                                Text(unit.label)
+                            }
+                        }
+                        .onChange(of: usngLabelUnit) { _ in
+                            (grid as! USNGGrid).labelUnit = usngLabelUnit
+                        }
+                    }
+                }
+            }
+            .navigationTitle("Grid Settings")
+            .navigationBarTitleDisplayMode(.inline)
+            .toolbar {
+                ToolbarItem(placement: .confirmationAction) {
+                    Button("Done") { dismiss() }
+                }
+            }
+            .onAppear {
+                updateUI()
+            }
+        }
+        
+        /// Update states and UI based on grid properties.
+        private func updateUI() {
+            gridIsVisible = grid.isVisible
+            if let lineSymbolColor = (grid.lineSymbols.first { $0 is LineSymbol } as? LineSymbol)?.color {
+                gridColor = Color(uiColor: lineSymbolColor)
+            }
+            gridType = GridType(grid: grid)!
+            labelsAreVisible = grid.labelsAreVisible
+            if let textSymbolColor = (grid.textSymbols.first { $0 is TextSymbol } as? TextSymbol)?.color {
+                labelsColor = Color(uiColor: textSymbolColor)
+            }
+            labelPosition = grid.labelPosition
+            if grid is LatitudeLongitudeGrid {
+                labelFormat = (grid as! LatitudeLongitudeGrid).labelFormat
+            } else if grid is MGRSGrid {
+                mgrsLabelUnit = (grid as! MGRSGrid).labelUnit
+            } else if grid is USNGGrid {
+                usngLabelUnit = (grid as! USNGGrid).labelUnit
+            }
+        }
+        
+        /// Changes the grid line color.
+        /// - Parameter color: The color for the grid lines.
+        private func changeGridColor(to color: UIColor) {
+            for symbol in grid.lineSymbols where symbol is LineSymbol {
+                let lineSymbol = symbol as! LineSymbol
+                lineSymbol.color = color
+            }
+        }
+        
+        /// Changes the grid labels color.
+        /// - Parameter color: The color for the text symbols.
+        private func changeLabelColor(to color: UIColor) {
+            for symbol in grid.textSymbols where symbol is TextSymbol {
+                let textSymbol = symbol as! TextSymbol
+                textSymbol.color = color
+            }
+        }
+    }
+}
+
+// MARK: - Helper Extensions
+
+private extension ShowGridView {
+    /// The kinds of grid to show in a map view.
+    enum GridType: CaseIterable {
+        case latLon, mgrs, usng, utm
+        
+        init?(grid: ArcGIS.Grid) {
+            switch grid {
+            case is LatitudeLongitudeGrid: self = .latLon
+            case is MGRSGrid: self = .mgrs
+            case is USNGGrid: self = .usng
+            case is UTMGrid: self = .utm
+            default: return nil
```

Comment:
Might be better to have the default be a fatalError rather than having a failable init because when you use the init you are force unwrapping anyways.